### PR TITLE
Fix Nullable and LowCardinality serialization for set index

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- Bug Fix

Short description (up to few sentences):
Fixed Set index for `Nullable` and `LowCardinality` columns. Before it, set index with `Nullable` or `LowCardinality` column led to error `Data type must be deserialized with multiple streams` while selecting.

### CODE REVIEW

File: dbms/src/Storages/MergeTree/MergeTreeSetSkippingIndex.cpp [hunk 1]
```diff
@@ -47,7 +47,16 @@ void MergeTreeSetIndexGranule::serializeBinary(WriteBuffer & ostr) const
     for (size_t i = 0; i < index.columns.size(); ++i)
     {
         const auto & type = index.data_types[i];
-        type->serializeBinaryBulk(*columns[i], ostr, 0, size());
+
+        IDataType::SerializeBinaryBulkSettings settings;
```
Comment (alexey-milovidov) [inline dbms/src/Storages/MergeTree/MergeTreeSetSkippingIndex.cpp]: Does it make sense to serialize DataTypeLowCardinality for index of type `set` as usual?

Comment (KochetovNicolai) [reply dbms/src/Storages/MergeTree/MergeTreeSetSkippingIndex.cpp]: Bulk serialization is needed for other types also. Nullable, for example.
For LowCardinality it doesn't (here we can only serialize dictionary, and even do not create set). But code will be a little bit more difficult.

Also, I don't understand why do we need to create set after deserialization.

Comment (alexey-milovidov) [reply dbms/src/Storages/MergeTree/MergeTreeSetSkippingIndex.cpp]: > Also, I don't understand why do we need to create set after deserialization.

For quick filtering of granules?

Comment (nikvas0) [reply dbms/src/Storages/MergeTree/MergeTreeSetSkippingIndex.cpp]: > Also, I don't understand why do we need to create set after deserialization.
Yes, we do not need set there, a simple block will be fine. I'll fix it later by extracting aggregation logic into another class.

**Pull request merged**